RecoTracker/LSTCore/src/alpaka/MiniDoublet.h

```diff
@@ -0,0 +1,1106 @@
+    }
+
+    pass = pass && (alpaka::math::abs(acc, dPhi) < miniCut);
+    if (not pass)
+      return pass;
+
+    // Cut #3: The dphi change going from lower Hit to upper Hit
+    // Ref to original code: https://github.com/slava77/cms-tkph2-ntuple/blob/184d2325147e6930030d3d1f780136bc2dd29ce6/doubletAnalysis.C#L3076
+    if (modulesInGPU.sides[lowerModuleIndex] != Center) {
+      // When it is tilted, use the new shifted positions
+      // TODO: This is somewhat of an mystery.... somewhat confused why this is the case
+      if (modulesInGPU.moduleLayerType[lowerModuleIndex] != SDL::Pixel) {
+        // dPhi Change should be calculated so that the upper hit has higher rt.
+        // In principle, this kind of check rt_lower < rt_upper should not be necessary because the hit shifting should have taken care of this.
+        // (i.e. the strip hit is shifted to be aligned in the line of sight from interaction point to pixel hit of PS module guaranteeing rt ordering)
+        // But I still placed this check for safety. (TODO: After checking explicitly if not needed remove later?)
+        // setdeltaPhiChange(lowerHit.rt() < upperHitMod.rt() ? lowerHit.deltaPhiChange(upperHitMod) : upperHitMod.deltaPhiChange(lowerHit));
+
+        dPhiChange = (rtLower < shiftedRt) ? SDL::deltaPhiChange(acc, xLower, yLower, shiftedX, shiftedY)
+                                           : SDL::deltaPhiChange(acc, shiftedX, shiftedY, xLower, yLower);
+        noShiftedDphiChange = rtLower < rtUpper ? SDL::deltaPhiChange(acc, xLower, yLower, xUpper, yUpper)
+                                                : SDL::deltaPhiChange(acc, xUpper, yUpper, xLower, yLower);
+      } else {
+        // dPhi Change should be calculated so that the upper hit has higher rt.
+        // In principle, this kind of check rt_lower < rt_upper should not be necessary because the hit shifting should have taken care of this.
+        // (i.e. the strip hit is shifted to be aligned in the line of sight from interaction point to pixel hit of PS module guaranteeing rt ordering)
+        // But I still placed this check for safety. (TODO: After checking explicitly if not needed remove later?)
+
+        dPhiChange = (shiftedRt < rtUpper) ? SDL::deltaPhiChange(acc, shiftedX, shiftedY, xUpper, yUpper)
+                                           : SDL::deltaPhiChange(acc, xUpper, yUpper, shiftedX, shiftedY);
+        noShiftedDphiChange = rtLower < rtUpper ? SDL::deltaPhiChange(acc, xLower, yLower, xUpper, yUpper)
+                                                : SDL::deltaPhiChange(acc, xUpper, yUpper, xLower, yLower);
+      }
+    } else {
+      // When it is flat lying module, whichever is the lowerSide will always have rt lower
+      dPhiChange = SDL::deltaPhiChange(acc, xLower, yLower, xUpper, yUpper);
+      noShiftedDphiChange = dPhiChange;
+    }
+
+    pass = pass && (alpaka::math::abs(acc, dPhiChange) < miniCut);
+    noShiftedDz = 0;  // not used anywhere
```

Comment:
Staying true to the inline comment, this variable is not used anywhere. Should I remove it from the code? Any clue by anyone if this will potentially fulfill a purpose and should be kept in? Maybe @slava77 has an idea why/how this was used in the past?